plugin.video.rivedila7/addon.py

```diff
@@ -12,97 +12,128 @@
-            a=div.a.get('href')
-            liStyle = xbmcgui.ListItem(dateRowWeek.contents[0]+" "+dateDay.contents[0]+" "+dateMonth.contents[0])
-            addDirectoryItem({"mode": "rivedi_la7","giorno": a}, liStyle)
-        xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
 
-def get_video_link(url,live):
-    req = urllib2.Request(url,headers=headers) 
-    page=urllib2.urlopen(req)
-    html=page.read();
-    if live!=True:
-        res=re.findall('mp4" : "(.*?)"', html)
-        if res:
-            return res[0]
-        else:
-            res=re.findall('mp4: "(.*?)"', html)
-            if res:
-                return res[0]
+def addDirectoryItem_nodup(parameters, li, title):
+    if title in list_programmi:
+        xbmc.log('Prog Duplicato',xbmc.LOGNOTICE)
     else:
-        res=re.findall('src: vS,//"(.*?)"', html)
-        if res:
-            return res[0]
+        url = sys.argv[0] + '?' + urllib.urlencode(parameters)
+        #xbmc.log('LIST: '+str(url),xbmc.LOGNOTICE)
+        return xbmcplugin.addDirectoryItem(handle=handle, url=url, listitem=li, isFolder=True)
 
-def play_video(video,live):
 
+def play_video(video,live):
+    link_video = ''
     if live:
         s = requests.Session()
         req = s.get(video,headers=headers)
         html = req.text
         vS = re.findall('var vS = \'(.*?)\';', html)
         try:
             link_video = vS[0]
-        except:
-            xbmc.log('[RivediLa7] Unable to find video url')
-    if not live and "tg.la7.it" in video:
-        req = urllib2.Request(video,headers=headers) 
+        except Exception as e:
+            e = sys.exc_info()[0]
+            xbmc.log('EXCEP VIDEO: '+str(e),xbmc.LOGNOTICE)
+            if xbmcgui.Dialog().ok(addon.getAddonInfo('name'), language(32005)):
+                return
+    else:
+        req = urllib2.Request(video,headers=headers)
         page=urllib2.urlopen(req)
-        html=BeautifulSoup(page,'html5lib')
-        if html.find("iframe"):
-            video=html.find("iframe")['src']
-    if not live and  "la7.it" in video:
-        link_video=get_video_link(video,live)
-    elif not live:
-        link_video=get_video_link(url_base+video,live)       
-
+        html=page.read();
+        res=re.findall('m3u8" : "(.*?)"', html)
+        if res:
+            link_video = res[0]
+        else:
+            res=re.findall('m3u8: "(.*?)"', html)
+            if res:
+                link_video = res[0]
+      
     listitem =xbmcgui.ListItem(titolo_global)
     listitem.setInfo('video', {'Title': titolo_global})
     if (thumb_global != ""):
         listitem.setArt({ 'thumb': thumb_global})
     listitem.setInfo('video', { 'plot': plot_global })
-    xbmc.Player().play(link_video, listitem)
+    if link_video == '':
+        xbmc.log('NO VIDEO LINK',xbmc.LOGNOTICE)
+        if xbmcgui.Dialog().ok(addon.getAddonInfo('name'), language(32005)):
+            return
+    else:
+        xbmc.Player().play(link_video, listitem)
```

Comment:
Plugins should not call `xbmc.Player()` directly. Note that by doing this, a new entry will be created on the database for each time you call the player. On the other hand, a plugin has a unique `plugin://` playable path which gets stored on the database and allows you to properly have resume points, resume status and watched history. The player class is supposed to be used by scripts when you need to do something a bit more complex (much of the time relying on skinning). The proper way to call the player in your addon is doing something like below:

```python
else:
    listitem.setPath(YOUR_PLAYABLE_PATH)
    xbmcplugin.setResolvedUrl(handle, True, listitem)
```